File: Producto/backend/src/main/java/backend/com/produccion/application/dto/RecepcionOCDTO.java
package backend.com.produccion.application.dto;

import jakarta.validation.Valid;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.util.List;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class RecepcionOCDTO {

    private Long idRecepcion;
    private Long ocId;
    private LocalDate fechaRecepcion;

    @Size(max = 50)
    private String numeroGuia;

    @Size(max = 100)
    private String responsable;

    private String observaciones;

    @Valid
    @NotEmpty(message = "Debe registrar al menos un item recibido")
    private List<RecepcionOCItemDTO> items;
}

File: Producto/frontend/src/remote/DTO/RecepcionOCDTO.js
import { RecepcionOCItemDTO } from './RecepcionOCItemDTO';

/**
 * DTO de Recepción de OC (ingreso de materiales).
 * Espejo de RecepcionOCDTO.java del backend.
 */
export class RecepcionOCDTO {
    constructor(data = {}) {
        this.idRecepcion = data.idRecepcion ?? null;
        this.ocId = data.ocId ?? null;
        this.fechaRecepcion = data.fechaRecepcion ?? null;
        this.numeroGuia = data.numeroGuia ?? '';
        this.responsable = data.responsable ?? '';
        this.observaciones = data.observaciones ?? '';
        this.items = Array.isArray(data.items)
            ? data.items.map(i => new RecepcionOCItemDTO(i))
            : [];
    }

    static fromResponse(response) {
        if (!response || !response.data) return null;
        return new RecepcionOCDTO(response.data);
    }

    static listFromResponse(response) {
        if (!response || !Array.isArray(response.data)) return [];
        return response.data.map(item => new RecepcionOCDTO(item));
    }

    get totalRecibido() {
        return this.items.reduce((acc, i) => acc + Number(i.cantidadRecibida || 0), 0);
    }
}
